src/battle-text-parser.ts

```diff
@@ -70,7 +76,9 @@ class BattleTextParser {
 				status = status.substr(1);
 			}
 		}
-		return {name, status, away};
+		// TODO: stop prefixing name with group and update callers
```

Comment:
It is very challenging to trace through not just the immediate callers of `parseNameParts` but the things they pass 'name' to etc, such that actually updating `name` to be just the name part and not group + name is out of scope for this PR (we can followup with a cleanup change to make people use `group` and  instead of all the places we use `test`/`chartAt(0)`/`substr(1)` and what not, but I'd really like to not block landing statuses on this).

Reply:
I'm fine with that.

Reply:
Wait, why are you manually adding a group to this function at all?

Reply:
This function needs to understand that names can be prefixed with a group to be able to properly parse text like `'@pre@!(Away) coding'`. Because `@` can be the moderator rank and the symbol used for separating name and status, the parsing will break if we don't handle the possibility. 

We could handle this by changing line 67 to just be `const atIndex = text.indexOf('@', 1);` instead (ie/ make use of the `fromIndex` to avoid getting choked up by the leading `@`), which would be totally valid. However, I left the actual group parsing in because `parseNameParts` seems like it really should, you know, parse the parts of the name. I'm fine to just use from index and not mess around with group at all in this method, but I believe in the future we'd probably want this method to be able to handle all parsing as mentioned in the first of these comments, its just annoying to do at this time.